formatter/src/main/java/org/togetherjava/formatter/CodeSectionFormatter.java

```diff
@@ -0,0 +1,396 @@
+        return type == TokenType.FOR && !internalEnhancedFor();
+    }
+
+    /**
+     * Checks if there's an enhanced for loop ahead (uses
+     * {@link SkippableLookaheadQueue#peek(int, Predicate)})
+     *
+     * @param type current token type
+     * @return wether there's an enhanced for loop or not
+     * @author illuminator3
+     */
+    private boolean isEnhancedForLoop(TokenType type) {
+        return type == TokenType.FOR && internalEnhancedFor();
+    }
+
+    /**
+     * Checks if there's an enhanced for loop ahead without checking the current token type
+     *
+     * @return wether there's an enhanced for loop ahead
+     * @author illuminator3
+     */
+    private boolean internalEnhancedFor() {
+        return queue.peek(3, t -> {
+            TokenType ttype = t.type();
+
+            return ttype == TokenType.ANNOTATION || ttype == TokenType.FINAL;
+        }).type() == TokenType.COLON;
+    }
+
+    /**
+     * Parenthesis rule: append a space after a closing parenthesis if the next token isn't another
+     * closing parenthesis, an operator or a semicolon
+     *
+     * @return wether a space should be put after the parenthesis
+     * @author illuminator3
+     */
+    private boolean isParenthesisRule(Token token) {
+        if (queue.isEmpty()) {
+            return false;
+        }
+
+        Token next = queue.peek();
+        TokenType nextType = next.type();
+
+        return token.type() == TokenType.CLOSE_PARENTHESIS
+                && nextType != TokenType.CLOSE_PARENTHESIS && nextType != TokenType.SEMICOLON
+                && !isOperator(next);
+    }
+
+    /**
+     * Appends a new line if there's more in the token queue
+     *
+     * @author illuminator3
+     */
+    private void appendNewLine() {
+        if (!queue.isEmpty()) {
+            if (forLevel <= 0) {
+                result.append('\n');
+            } else {
+                result.append(' '); // put a space after a ; in a for loop
+            }
+        }
+    }
+
+    /**
+     * Checks if a given token is a keyword using {@link TokenType#isKeyword()}
+     *
+     * @param token token to check
+     * @return wether the given token is a keyword
+     * @author illuminator3
+     */
+    private boolean isKeyword(Token token) {
+        return token.type().isKeyword();
+    }
+
+    /**
+     * Checks if a given token is an operator using {@link TokenType#isOperator()}
+     *
+     * @param token token to check
+     * @return wether the given token is an operator
+     * @author illuminator3
+     */
+    private boolean isOperator(Token token) {
+        return token.type().isOperator();
+    }
+
+    /**
+     * Updates the indentation based on the current token type
+     *
+     * @param type current token type
+     * @author illuminator3
+     */
+    private void updateIndentation(TokenType type) {
+        switch (type) {
+            case OPEN_BRACES -> indentation++;
+            case CLOSE_BRACES -> indentation--;
```

Comment:
I did it like that, since I don't want to do
```java
if (token.type() == TokenType.OPEN_BRACES || token.type() == TokenType.CLOSE_BRACES) {
    updateIndentation(token.type());
}
```
everywhere, so I just ignore anything other than those who are actually used.

Reply:
The switch is fine, it's missing a default case tho. ^^

Something like `default -> throw new IllegalStateException("Invalid token type " + type) `
Or at least a logger if you don't want to throw.

Reply:
The method is also called if the type is anything other than those two, so throwing an exception wouldn't make any sense.

Reply:
sure, then log it ^^

Reply:
This is a critical issue in sonar, which means it wouldn't let you merge.

Reply:
> This is a critical issue in sonar

![image](https://user-images.githubusercontent.com/49957334/133297299-72f3d2b8-fb63-4021-9ff0-b264eaf9b4fd.png)

Reply:
I don't see the critical issue in sonar, only issue I can see is incorrectly formatted

Reply:
Logging it would also not make sense, since again, it's not an error.

Reply:
sure, then add the comment explaining that ^^

Reply:
![image](https://user-images.githubusercontent.com/30927961/133298296-aa07ec10-7d37-4cf2-be78-0d71e98a3c1d.png)

Reply:
I'll change it into an if-elseif to satisfy sonar.